feat(blog): add blog for September (#53)
mun-lang/website: content/posts/2020-09-30-this-month-september.md

---
title: "This Month in Mun - September 2020"
author: "The Mun Team"

> excerpt: "The community and core team worked together on improvements to bring Mun closer to a new release."

Reviewer: Maybe add the last sentence of the first paragraph to the excerpt to. For some more body and such they match.

---

The community and core team worked together on improvements to bring Mun closer to a new release. 
Work is also progressing to support multiple source files.

#### Community

A community member whom we welcomed last month picked up another [*good first issue*][gfi] and also 
fixed some deprecations:

* **build: pin rust version to 1.46.0** [[PR#269]](https://github.com/mun-lang/mun/pull/269)
* **improvement: bump smol_str dependency to 0.1.17** [[PR#270]](https://github.com/mun-lang/mun/pull/270)

We are very thankful for your contributions and we hope to see more of them in the future!

> More community contributions came from a contributor who tried out our C++ bindings and who 

Reviewer: ```suggestion
More community contributions came from a contributor who tried out our C++ bindings and
```

This includes some bug fixes which were found by the Mun community as well as more [high level 
documentation about our use of Salsa](https://docs.mun-lang.org/ch04-01-salsa.html).

* **fix: never return type in let initializer** [[PR#264]](https://github.com/mun-lang/mun/pull/264)
* **refactor: use Idx<T> instead of macro in arena** [[PR#266]](https://github.com/mun-lang/mun/pull/266)
* **misc: split database and added docs** [[PR#267]](https://github.com/mun-lang/mun/pull/267)

Work is still underway to support multiple files in a single project but supporting PRs have already 
landed.

* **feature: adds fixtures to support multiple files from string** [[PR#272]](https://github.com/mun-lang/mun/pull/272)
* **feature: refactored RawItems into ItemTree** [[PR#271]](https://github.com/mun-lang/mun/pull/271)

With the work done for project configuration [in July](../../../07/30/this-month-july/), we are 
focusing on compiler support for [`use` statements](https://github.com/mun-lang/mun/issues/249) and 
[accessibility specifiers](https://github.com/mun-lang/mun/issues/248).

Last month we talked about wanting to release Mun v0.3 but we are not quite there yet. We chose to 

> delay the release a little bit to give ourselves some time to properly focus on polish and get to 

Reviewer: ```suggestion
delay the release a little bit to give ourselves some time to achieve the level of polish that we
aspire to deliver in every release.
```

> the quality standard we aspire. 

Reviewer: ```suggestion
```
